src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -513,11 +513,11 @@ private async Task<bool> HashFilesAndPutInAnalysisQueueAsnc()
 
                         loggerCache ??= new Dictionary<string, CachingLogger>();
 
-                        if (!loggerCache.TryGetValue(hashData.Sha256, out CachingLogger logger))
-                        {
-                            logger = loggerCache[hashData.Sha256] = (CachingLogger)context.Logger;
-                        }
-                        context.Logger = logger;
+                        context.Logger = hashData?.Sha256 == null
+                            ? (CachingLogger)context.Logger
```

Comment:
>? (CachingLogger)context.Logger [](http://example.com/codeflow?start=28&length=31)

if no hash, have to set the context.Logger to itself? can we skip in this case?

Reply:
Yes, let's just wrap this code in a check that hashData.Sha256 is non-null, will be more readable.

Could we get this change in and then also try to mock up a test for this condition? 

I'm concerned we're not necessarily handling this negative condition comprehensively.

Reply:
Could we get this change in and then also try to mock up a test for this condition? ---- done fixing the code, will need to mock up a test.